datetime.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+
+package gosnowflake
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var incorrectSecondsFractionRegex = regexp.MustCompile(`[^.,]FF`)
+var correctSecondsFractionRegex = regexp.MustCompile(`FF(?P<fraction>\d?)`)
+
+type formatReplacement struct {
+	input  string
+	output string
+}
+
+var formatReplacements = []formatReplacement{
+	{input: "YYYY", output: "2006"},
+	{input: "YY", output: "06"},
+	{input: "MMMM", output: "January"},
+	{input: "MM", output: "01"},
+	{input: "MON", output: "Jan"},
+	{input: "DD", output: "02"},
+	{input: "DY", output: "Mon"},
+	{input: "HH24", output: "15"},
+	{input: "HH12", output: "03"},
+	{input: "AM", output: "PM"},
+	{input: "MI", output: "04"},
+	{input: "SS", output: "05"},
+	{input: "TZH", output: "Z07"},
+	{input: "TZM", output: "00"},
+}
+
+func snowflakeFormatToGoFormat(sfFormat string) (string, error) {
+	res := sfFormat
+	for _, replacement := range formatReplacements {
```

Comment:
If the format is `DD/MM/YY` it can be parsed correctly, but it will lead to incorrect time. I wonder if we can verify it somehow.